src/plugins/data/public/ui/filter_bar/filter_options.tsx

```diff
@@ -29,177 +29,401 @@
+            closePopover();
+            onRemoveAll();
           },
           'data-test-subj': 'removeAllFilters',
+          disabled: disableMenuOption,
+          className: useNewHeader ? 'globalFilterGroup__removeAllFilters' : '',
         },
       ],
-    };
-
-    return (
-      <EuiPopover
-        id="popoverForAllFilters"
-        className="globalFilterGroup__allFiltersPopover"
-        isOpen={this.state.isPopoverOpen}
-        closePopover={this.closePopover}
-        button={
-          <EuiToolTip
-            content={this.props.intl.formatMessage({
-              id: 'data.filter.options.changeAllFiltersButtonLabel',
-              defaultMessage: 'Change all filters',
-            })}
-            delay="long"
-            position="bottom"
-          >
-            <EuiSmallButtonIcon
-              onClick={this.togglePopover}
-              iconType="filter"
-              aria-label={this.props.intl.formatMessage({
-                id: 'data.filter.options.changeAllFiltersButtonLabel',
-                defaultMessage: 'Change all filters',
-              })}
-              data-test-subj="showFilterActions"
-            />
-          </EuiToolTip>
-        }
-        anchorPosition="rightUp"
-        panelPaddingSize="none"
-        repositionOnScroll
-      >
-        <EuiPopoverTitle paddingSize="s">
-          <FormattedMessage
-            id="data.filter.searchBar.changeAllFiltersTitle"
-            defaultMessage="Change all filters"
-          />
-        </EuiPopoverTitle>
-        <EuiContextMenu initialPanelId={0} panels={[panelTree]} size="s" />
-      </EuiPopover>
-    );
+    },
+  ];
+
+  const handleSave = () => {
+    if (props.onSave) {
+      props.onSave();
+    }
+    setIsPopoverOpen(false);
+  };
+
+  const saveQueryPanel = (
+    <EuiContextMenuPanel
+      items={[
+        <SavedQueryManagementComponent
+          showSaveQuery={props.showSaveQuery}
+          loadedSavedQuery={props.loadedSavedQuery}
+          onSave={handleSave}
+          onSaveAsNew={props.onSaveAsNew!}
+          onLoad={props.onLoad!}
+          savedQueryService={props.savedQueryService!}
+          onClearSavedQuery={props.onClearSavedQuery!}
+          closeMenuPopover={() => {
+            setIsPopoverOpen(false);
+          }}
+          key={'savedQueryManagement'}
+        />,
+      ]}
+      data-test-subj="save-query-panel"
+    />
+  );
+
+  const menuPanel = (
+    <EuiContextMenu
+      initialPanelId={0}
+      panels={panelTree}
+      size="s"
+      data-test-subj="filter-options-menu-panel"
+    />
+  );
+  const addFilterPanel = (
+    <EuiContextMenuPanel
+      items={[
+        <EuiResizeObserver onResize={onResize} key={'filter-option-resize'}>
+          {(resizeRef) => (
+            <div style={{ width: maxFilterWidth, maxWidth: '100%' }} ref={resizeRef}>
+              <EuiFlexItem style={{ width: filterWidth }} grow={false}>
+                <FilterEditor
```

Comment:
Is there any value to defining this here instead of within your case statement below?